### DISCUSSION TITLE
PID regulation on Burner value  0=Auto, 1=0%, 101=100% - (Don't want 0=Auto)

### DISCUSSION THREAD

**Discussion by bigl00z3** [artisan-roaster-scope/artisan#819] [Q&A] [answered]
Using the Artisan internal PID  I ma having issues trying to regulate the temperature by controling the power.

The equipment is set as follow ( I can't modify this behaviour):

- Value 0 = power "automatic mode" = controled by the roaster, not artisan, I don't want this
- Value 1 =  at 0% power
- Value 2 = at 1% Power
- ... all the way to:
- Value 101= at 100% Power

This creates issues when the PID tries to stop the power (set to 0%) but instead switches to "Auto mode"

I have tried to set the min to 1%, but it didn't seem to work.
![image](https://user-images.githubusercontent.com/8886922/153430208-4d5f14a2-3c18-4579-bdea-3f735ef16b6b.png)

Any idea? 

[artisan-settings.zip](https://github.com/artisan-roaster-scope/artisan/files/8041866/artisan-settings.zip)

**Comment by MAKOMO** [comment]
Nico, this looks like a Giesen setup. The burner slider has an offset of 1 thus setting the slider to 0 should emit a 1 avoiding your auto mode, right?

**Reply by bigl00z3** [reply]
You are correct, there is the offset. And further tests showed that it wasn't in auto mode.

It's the internal PID that suddenly stops regulating. Locks itself with a positive value and does not change anymore.
Stoping the PID with `PIDoff` followed directly by a `PIDon` starts regulating again.

I created a "rollercoaster" designer template, ran the empty roaster (note the P,I,D values are not nice at present point) , and you can see when BT doesn't follow the Background BT, the events are marked when I do PIDon/off as red event markers:

<img width="1075" alt="image" src="https://user-images.githubusercontent.com/8886922/153452269-8c7a4c69-ba72-482b-ab39-2cf800e6343b.png">

**Comment by MAKOMO** [comment, marked as answer]
Machine seems to heat up at one point and not following the last duty value send by the Artisan PID. Not much to do on the Artisan side about it.
